pkg/api/types.go

```diff
@@ -758,12 +758,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
It's not enforced, but I'd like as much consistency as possible in v1beta3.

At the moment, is it possible for this list to contain hostnames? Besides, I don't think "Address" is appropriate for a hostname.